keepalived-vip/README.md

```diff
@@ -0,0 +1,374 @@
+# kube-keepalived-vip
+Kubernetes Virtual IP address/es using [keepalived](http://www.keepalived.org)
+
+AKA "how to set up virtual IP addresses in kubernetes using [IPVS - The Linux Virtual Server Project](http://www.linuxvirtualserver.org/software/ipvs.html)".
+
+## Disclaimer:
+- This is a **work in progress**.
+
+## Overview
+
+There are 2 ways to expose a service in the current kubernetes service model:
+
+- Create a cloud load balancer.
+- Allocate a port (the same port) on every node in your cluster and proxy traffic through that port to the endpoints.
+
+This just works. What's the issue then? 
+
+The issue is that it does not provide High Availability because beforehand is required to know the IP addresss of the node where is running and in case of a failure the pod can be be moved to a different node. Here is where ipvs could help. 
+The idea is to define an IP address per service to expose it outside the Kubernetes cluster and use vrrp to announce this "mapping" in the local network.
+With 2 or more instance of the pod running in the cluster is possible to provide high availabity using a single IP address.
+
+##### What is the difference between this and [service-loadbalancer](https://github.com/kubernetes/contrib/tree/master/service-loadbalancer) or [nginx-alpha](https://github.com/kubernetes/contrib/tree/master/Ingress/controllers/nginx-alpha) to expose one or more services?
+
+This should be considered a complement, not a replacement for HAProxy or nginx. The goal using keepalived is to provide high availability and to bring certainty about how an exposed service can be reached. For instance keepalived can use used to expose the service-loadbalancer or nginx ingress controller in the LAN using one IP address.
+
```

Comment:
Please also add a section clarifying the difference between this and simply giving service-loadbalancer a vip via a nodeport service (the main difference would probably be that the vrrp vip is discoverable).